staging/src/k8s.io/apiserver/pkg/endpoints/responsewriter/testing/recorder.go

```diff
@@ -0,0 +1,31 @@
+/*
+Copyright 2024 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package testing
+
+import (
+	"strings"
+)
+
+func IsStreamReadOrWriteTimeout(err error) bool {
+	if err == nil {
+		return false
+	}
+	// we get the the following stream reset error due to a
+	// timeout from the per handler write timeout.
+	return strings.Contains(err.Error(), "stream error: stream ID") &&
+		strings.Contains(err.Error(), "INTERNAL_ERROR")
```

Comment:
helpful to add a small unit test to verify that these error strings don't change?

Reply:
yes, added